gradio_app.py

```
@@ -0,0 +1,1258 @@
+            val_dataset = EnhancedWakewordDataset(
+                wakeword_val, negative_val[:len(negative_val)//2],
+                negative_val[len(negative_val)//2:], background_files[:50],
+                self.processor, augment=False
+            )
+
+            # Create dataloaders
+            # On Windows, multi-processing workers can cause storage resize errors with numpy/librosa.
+            # Use single-process loading there for stability.
+            import os as _os
+            _num_workers = 0 if _os.name == 'nt' else 2
+            self.train_loader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True, num_workers=_num_workers, collate_fn=collate_fn)
```

Ruff (0.13.1)

Undefined name `collate_fn`

(F821)